### PR TITLE
Modularize policy controller

### PR DESCRIPTION
**Pull request by adleong**
This change improves the modularity of the policy controller by splitting code related in inbound policy and outbound policy into separate modules.  There are no functional changes in this PR, only moves and renames.  We have moved any code which was used by both inbound and outbound into common modules so that there are no cross dependencies between inbound and outbound.

Additionally, several types that had names beginning with "Inbound" or "Outbound" have had this prefix removed since they are now distinguishable by their modules.  e.g. `InboundHttpRoute` becomes `inbound::HttpRoute`.

### CODE REVIEW

File: policy-controller/k8s/index/src/outbound/mod.rs
Comment (alpeb) [inline policy-controller/k8s/index/src/outbound/mod.rs:1]: I might be off, but I think the new recommended style is to put this in `policy-controller/k8s/index/src/outbound.rs`?

**Review by alpeb — APPROVED** [review]
Looks good to me modulo module comment :drum:

File: policy-controller/k8s/index/src/outbound/mod.rs
Comment (olix0r) [reply policy-controller/k8s/index/src/outbound/mod.rs:1]: Yeah, we do generally prefer these new-style modules, as it makes it trivial to turn any file into a parent module without renames etc.

File: policy-controller/k8s/index/src/pod.rs
Comment (olix0r) [inline policy-controller/k8s/index/src/pod.rs:1]: Should this be in the `inbound` module?

File: policy-controller/k8s/index/src/tests.rs
Comment (olix0r) [inline policy-controller/k8s/index/src/tests.rs:1]: Should this be moved into the `inbound` module?

**Review by olix0r — COMMENTED** [review]
generally i'm into this! it seems like there's a few places where we should move more existing things into the inbound module

File: policy-controller/k8s/index/src/pod.rs
Comment (adleong) [reply policy-controller/k8s/index/src/pod.rs:1]: We use the `PortSet` type in the outbound index for opaque ports.  I also anticipate needing more pod metadata functionality in the outbound in the future as we add support for endpoint backends.

File: policy-controller/core/src/inbound.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    /// This is required for ordering returned `HttpRoute`s by their creation
+    /// timestamp.
+    pub creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct HttpRouteRule {
+    pub matches: Vec<HttpRouteMatch>,
+    pub filters: Vec<Filter>,
+}
```
[the remark this answers is not in the document]
Comment (adleong) [reply policy-controller/core/src/inbound.rs:110]: Yeah possibly, but I'm inclined to wait until there's more duplication before trying to factor it out.

**Pull request merged**